### ISSUE TITLE
example input videos

### ISSUE THREAD

**Issue by jimmyl02** [alimama-creative/M3DDM-Video-Outpainting#2]
Hello,

Thanks for the amazing work! I was wondering if you have some sample videos that we could use with the inference script to play around with. I haven't had the chance to read through the code yet but I'm guessing the model was trained with a certain aspect ratio and resolution. Would love to be able to recreate some of the examples from the landing page.

Thanks! Would appreciate any insights.

**Comment by fanfanda** [comment]
Our M3DDM can accept videos of any resolution along with a target_ratio_list, but the resolution of the longest edge of the output video will be 256. This is because our training dataset was resized to a resolution of 256x256. Additionally, the target_ratio_list that is input must differ from the input video's aspect ratio to be meaningful. You can download videos from the YouTube-VOS dataset and then use our script with a target_ratio_list of 1:1 for video extension. Here is an example generated using the video 13006c4c7e.mp4 from YouTube-VOS with a target_ratio_list of 1:1.

https://github.com/alimama-creative/M3DDM-Video-Outpainting/assets/25380176/2a77d955-3eaa-40fd-bace-4ceaf56e2a1e

![genvideo](https://github.com/alimama-creative/M3DDM-Video-Outpainting/assets/25380176/c1e571fb-47d6-4243-93d8-ccd539b29dbf)

**Comment by fanfanda** [comment]
Our M3DDM can take input videos of any resolution and any target_ratio_list, but the output video will have its longest edge at a resolution of 256. This is due to our training dataset being resized to a 256x256 resolution. Furthermore, the input target_ratio_list must be different from the aspect ratio of the input video, as this is what makes the process meaningful. You can download videos from the YouTube-VOS dataset and use our script with a 1:1 target_ratio_list to extend the video. Below is an example created using the video 13006c4c7e.mp4 from YouTube-VOS, with a 1:1 target_ratio_list.




| Source Video | Output Result |
|--------------|---------------|
| [Video1](https://github.com/alimama-creative/M3DDM-Video-Outpainting/assets/25380176/c5ca5e73-1e5a-47dd-999d-e9c9318f9e64)       | ![genvideo](https://github.com/alimama-creative/M3DDM-Video-Outpainting/assets/25380176/737e5c9a-0ce8-4f29-b6a9-6fa0c2c6529a)       |

**Comment by iGerman00** [comment]
It's a very interesting paper. With some additional work such as masking in post to remove any really annoying flashing and temporal inconsistency, color correction to match the original high-resolution video, as well as an upscale with a model like Real-ESRGAN or Topaz' products, I was able to composite an extension for an iconic 4:3 video to 16:9. I used the pre-trained weights provided in the README. Took very long on a 3090, though, about 10 hours for this video.


https://github.com/alimama-creative/M3DDM-Video-Outpainting/assets/36676880/f64faede-6bc3-41a3-9523-abc96dac9b7e

**Comment by fanfanda** [comment]
Wow, this is a really nice result! Thanks for your contributions.

We are also continuing to optimize our video outpainting model. In the future, we will support output for high-resolution videos without the need for subsequent super-resolution models. Regarding inference speed, currently you can manually remove the branch for global video frames, or use a stride of [15, 1] instead of [15, 5, 1] for a speed-up, though be aware that this might result in some loss of quality. Alternatively, you could use existing distillation algorithms to speed up our model.
